llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -36980,6 +36980,18 @@ void X86TargetLowering::computeKnownBitsForTargetNode(const SDValue Op,
     }
     break;
   }
+  case ISD::INTRINSIC_WO_CHAIN: {
+    switch (Op->getConstantOperandVal(0)) {
+    case Intrinsic::x86_sse2_psad_bw:
+    case Intrinsic::x86_avx2_psad_bw:
+    case Intrinsic::x86_avx512_psad_bw_512:
+      // PSADBW - fills low 16 bits and zeros upper 48 bits of each i64 result.
+      assert(VT.getScalarType() == MVT::i64 && "Unexpected PSADBW types");
+      Known.Zero.setBitsFrom(16);
+      break;
```

Comment:
This is obv improvement as is, but why did you drop your impl:
https://github.com/llvm/llvm-project/issues/81765#issuecomment-1954721173

Reply:
It just a question of what order we put these patches in - if #81765 gets completed first then I'll just update this one to match

Reply:
Why not just write a helper that does the right logic now, then your patch for https://github.com/llvm/llvm-project/issues/81765 will just be calling the function for `X86ISD::PSADBW` as well?

Don't feel strongly, just seems strange.

Reply:
I know its documented as being 16 bits, but its worst case 11 bits I think. 255*8 fits in 11 bits.